Detectors/MUON/MCH/Align/src/AlignRecordSpec.cxx

```diff
@@ -0,0 +1,369 @@
+#include "DetectorsCommonDataFormats/AlignParam.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "DetectorsCommonDataFormats/DetectorNameConf.h"
+#include "MCHGeometryTransformer/Transformations.h"
+#include "ReconstructionDataFormats/GlobalFwdTrack.h"
+#include "ReconstructionDataFormats/GlobalTrackID.h"
+#include "ReconstructionDataFormats/PrimaryVertex.h"
+#include "ReconstructionDataFormats/StrangeTrack.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "ReconstructionDataFormats/TrackMCHMID.h"
+
+#include <TTree.h>
+#include <TChain.h>
+
+const int fgNCh = 10;
+const int fgNDetElemCh[fgNCh] = {4, 4, 4, 4, 18, 18, 26, 26, 26, 26};
+const int fgSNDetElemCh[fgNCh + 1] = {0, 4, 8, 12, 16, 34, 52, 78, 104, 130, 156};
+
+namespace o2
+{
+namespace mch
+{
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+using namespace std;
+using std::cout;
+using std::endl;
+using DataRequest = o2::globaltracking::DataRequest;
+using GID = o2::dataformats::GlobalTrackID;
+
+class AlignRecordTask
+{
+ public:
+  //_________________________________________________________________________________________________
+  AlignRecordTask(std::shared_ptr<DataRequest> dataRequest, std::shared_ptr<base::GRPGeomRequest> ccdbRequest, bool useMC = true)
+    : mDataRequest(dataRequest), mCCDBRequest(ccdbRequest), mUseMC(useMC) {}
+
+  //_________________________________________________________________________________________________
+  void init(framework::InitContext& ic)
+  {
+
+    LOG(info) << "initializing align record maker";
+    mTrackCount = 0;
+    mTrackMatched = 0;
+    mTrackMatchedPerTF = 0;
+    if (mCCDBRequest) {
+      base::GRPGeomHelper::instance().setRequest(mCCDBRequest);
+    } else {
+      auto grpFile = ic.options().get<std::string>("grp-file");
+      if (std::filesystem::exists(grpFile)) {
+        const auto grp = parameters::GRPObject::loadFrom(grpFile);
+        base::Propagator::initFieldFromGRP(grp);
+        TrackExtrap::setField();
+        TrackExtrap::useExtrapV2();
+        mAlign.SetBFieldOn(mch::TrackExtrap::isFieldON());
+      } else {
+        LOG(fatal) << "GRP file doesn't exist!";
+      }
+    }
+
+    // Configuration for alignment object
+    mAlign.SetDoEvaluation(false);
+    mAlign.DisableRecordWriter();
+    mAlign.SetAllowedVariation(0, 2.0);
+    mAlign.SetAllowedVariation(1, 0.3);
+    mAlign.SetAllowedVariation(2, 0.002);
+    mAlign.SetAllowedVariation(3, 2.0);
+
+    // Configuration for track fitter
+    const auto& trackerParam = TrackerParam::Instance();
+    trackFitter.setBendingVertexDispersion(trackerParam.bendingVertexDispersion);
+    trackFitter.setChamberResolution(trackerParam.chamberResolutionX, trackerParam.chamberResolutionY);
+    trackFitter.smoothTracks(true);
+    trackFitter.useChamberResolution();
+    mImproveCutChi2 = trackerParam.sigmaCutForImprovement;
```

Comment:
the chi2 cut is defined as:
`mImproveCutChi2 = 2. * trackerParam.sigmaCutForImprovement * trackerParam.sigmaCutForImprovement;`

Reply:
Thanks for pointing this out.